packages/workspace/src/browser/workspace-delete-handler.ts

```diff
@@ -158,4 +158,17 @@ export class WorkspaceDeleteHandler implements UriCommandHandler<URI[]> {
         await Promise.all(pending);
     }
 
+    /**
+     * Perform deletion of multiple URIs.
+     *
+     * @param uris URIs of selected resources.
+     */
+    protected async deleteAll(uris: URI[]): Promise<void> {
+        await this.workspaceService.deleteFiles(uris, (error, file) => {
```

Comment:
Why do we do it? We already have central logic to deal with deletion events. We should react on change from `FileSystem`, don't do it only sometimes for one page.

Reply:
If you look at VS Code's documentation, the do mention that these events don't come from fs but from "user gestures" (https://code.visualstudio.com/api/references/vscode-api#workspace).

> Note 1: This event is triggered by user gestures, like deleting a file from the explorer, or from the workspace.applyEdit-api, but this event is not fired when files change on disk, e.g triggered by another application, or when using the workspace.fs-api.

Reply:
Think about it: the `will` events are designed with a `waitUntil` mechanism, file system events never provide a way to catch when a file _will_ be deleted, we only react after it was done. So I think if an extension wants to listen for fs events it should use the fs watch apis?

Reply:
I did not meant file watching (nsfw), I meant `FileSystem` API which is used by frontend widgets: https://github.com/eclipse-theia/theia/pull/7718#issuecomment-622424019

Reply:
Also closing is not correct. Deleted widgets should be marked as deleted and we have the code doing it here: https://github.com/eclipse-theia/theia/blob/7498bd1744fb00e583a8712220bfa615e54c669e/packages/filesystem/src/browser/filesystem-frontend-contribution.ts#L235

Reply:
I cannot reproduce?